Userland/Applications/FileManager/PropertiesWindow.cpp

```diff
@@ -172,6 +173,16 @@ ErrorOr<void> PropertiesWindow::create_widgets(bool disable_rename)
     m_apply_button->on_click = [this](auto) { apply_changes(); };
     m_apply_button->set_enabled(false);
 
+    if (S_ISDIR(m_old_mode)) {
+        m_directory_statistics_calculator = make_ref_counted<DirectoryStatisticsCalculator>(m_path);
+        m_directory_statistics_calculator->on_update = [this, origin_event_loop = &Core::EventLoop::current()](off_t total_size_in_bytes, size_t file_count, size_t directory_count) {
+            origin_event_loop->deferred_invoke([=, this] {
+                m_size_label->set_text(DeprecatedString::formatted("{}\n{} files, {} subdirectories", human_readable_size_long(total_size_in_bytes), file_count, directory_count));
```

Comment:
what happens if the properties window is closed while this cross-thread event is in the pipe?

Feels like if the system is under significant CPU pressure we could have a race condition here and use-after-free on the properties window.

If the properties window is not already Weakable, it should be, and we should grab a weak reference to `this` here to check if `this` is actually alive. Search for `strong_this` in the codebase for the pattern.

Reply:
I've reworked this to use a weak pointer. 

I did see other places in the codebase use:

`deferred_invoke([strong_this = NonnullRefPtr(*this)]() {})`

I assume what I have is preferable though, as there is no guarantee the lambda will be executed?